src/Tribe/Editor/Utils.php

```diff
@@ -41,25 +41,53 @@ public function to_block_name( $name = '' ) {
 	 *
 	 * @param        $post_id
 	 * @param string $block_name
+	 * @param string $replacement
 	 *
 	 * @return bool
 	 */
-	public function remove_block( $post_id, $block_name = '' ) {
+	public function remove_block( $post_id, $block_name = '', $replacement = '' ) {
+		$patttern = '/^\s*<!-- ' . $block_name . '.*\/-->\s*$/gm';
+		return $this->update_post_content( $post_id, $patttern, $replacement );
+	}
+
+	/**
+	 * Function used to remove the inner blocks and the parent block as well inside of a post_content
+	 *
+	 * @since TBD
+	 *
+	 * @param        $post_id
+	 * @param        $block_name The name of the block
+	 * @param string $replacement The string used to replace the value of the searched block
+	 *
+	 * @return bool
+	 */
+	public function remove_inner_blocks( $post_id, $block_name, $replacement = '' ) {
+		$pattern = '/^\s*<!-- ' . $block_name . '.*-->\s.*<!-- \/' . $block_name . ' -->/ms';
+		return $this->update_post_content( $post_id, $pattern, $replacement );
+	}
+
+	/**
+	 * Update the content of a post using a pattern to search a specifc string, with a custom
+	 * replacement
+	 *
+	 * @since TBD
+	 *
+	 * @param        $post_id
+	 * @param        $pattern
+	 * @param string $replacement The string used to replace the value of the searched block
+	 *
+	 * @return bool
+	 */
+	public function update_post_content( $post_id, $pattern, $replacement = '' ) {
```

Comment:
Also make this Private method, since it should only be accessed by the other method.

Reply:
I think it might be helpful for an outsider that wants to execute an update on the `post_content` as the only params are: 

- `$post_id`
- `$pattern`
- `$replacement` (optional)

I think `public` makes sense as will provide externals just to call `tribe( 'editor' )->update_post_content( $post_id, $pattern )` to trigger an update on post content. 

Also as does not modifies any of the internals of the class I don't think making `public` is a bad thing as does not exposes any internal variables as it uses `wp_update_post` under the hood to do it 

Let me know what do you think about this.